recipes/amgcl/all/conanfile.py

```diff
@@ -20,12 +21,19 @@ class AmgclConan(ConanFile):
     package_type = "header-library"
     settings = "os", "arch", "compiler", "build_type"
     no_copy_source = True
+    options = { 'with_boost': [True, False] }
+    default_options = { 'with_boost': True }
+
+    def config_options(self):
+        if Version(self.version) < "1.4.4":
+            del self.options.with_boost
 
     def layout(self):
         basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("boost/1.81.0")
+        if self.options.get_safe("with_boost"):
+            self.requires("boost/1.81.0")
```

Comment:
```suggestion
            self.requires("boost/1.85.0")
```

Less conflicts. As you suggested over on slack, a range would be a solution if it wasn't boost, it's too big to confidentely add ranges to it unconditionally. Maybe for header-only libraries it would be a valid approach?